operator/internal/common/extensions/validate_deployment_defaults_test.go

```diff
@@ -0,0 +1,112 @@
+	tests := map[string]struct {
+		obj         *unstructured.Unstructured
+		expectError bool
+	}{
+		"valid configuration": {
+			obj: toUnstructured(t, centralWithDeploymentDefaults(&platform.DeploymentDefaultsSpec{
+				NodeSelector: map[string]string{
+					"global-node-selector-label": "global-node-selector-value",
+				},
+				Tolerations: []*corev1.Toleration{
+					{Key: "node.stackrox.io", Value: "false", Operator: corev1.TolerationOpEqual},
+				},
+			})),
+			expectError: false,
+		},
+		"invalid: pinToNodes with nodeSelector": {
+			obj: toUnstructured(t, centralWithDeploymentDefaults(&platform.DeploymentDefaultsSpec{
+				PinToNodes: ptr.To(platform.PinToNodesInfraRole),
+				NodeSelector: map[string]string{
+					"global-node-selector-label": "global-node-selector-value",
+				},
+			})),
+			expectError: true,
+		},
+		"invalid: pinToNodes with tolerations": {
+			obj: toUnstructured(t, centralWithDeploymentDefaults(&platform.DeploymentDefaultsSpec{
+				PinToNodes: ptr.To(platform.PinToNodesInfraRole),
+				Tolerations: []*corev1.Toleration{
+					{Key: "node.stackrox.io", Value: "false", Operator: corev1.TolerationOpEqual},
+				},
+			})),
+			expectError: true,
+		},
+		"invalid: pinToNodes with both nodeSelector and tolerations": {
+			obj: toUnstructured(t, centralWithDeploymentDefaults(&platform.DeploymentDefaultsSpec{
+				PinToNodes: ptr.To(platform.PinToNodesInfraRole),
+				NodeSelector: map[string]string{
+					"global-node-selector-label": "global-node-selector-value",
+				},
+				Tolerations: []*corev1.Toleration{
+					{Key: "node.stackrox.io", Value: "false", Operator: corev1.TolerationOpEqual},
+				},
+			})),
+			expectError: true,
+		},
+		"skips validation for deleted objects": {
+			obj: func() *unstructured.Unstructured {
+				central := centralWithDeploymentDefaults(&platform.DeploymentDefaultsSpec{
+					PinToNodes: ptr.To(platform.PinToNodesInfraRole),
+					NodeSelector: map[string]string{
+						"global-node-selector-label": "global-node-selector-value",
+					},
+				})
+				now := metav1.NewTime(time.Now())
+				central.SetDeletionTimestamp(&now)
+				return toUnstructured(t, central)
+			}(),
+			expectError: false,
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			ext := ValidateDeploymentDefaultsExtension()
+			err := ext(context.Background(), tt.obj, nil, logr.Discard())
+
+			if tt.expectError {
+				assert.Error(t, err, "expected validation to fail")
+			} else {
+				assert.NoError(t, err, "expected validation to pass")
+			}
+		})
+	}
+}
+
+func centralWithDeploymentDefaults(d *platform.DeploymentDefaultsSpec) *platform.Central {
+	return &platform.Central{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "stackrox",
+		},
+		Spec: platform.CentralSpec{
+			Customize: &platform.CustomizeSpec{
+				DeploymentDefaults: d,
+			},
+		},
+	}
+}
+
+func toUnstructured(t *testing.T, obj runtime.Object) *unstructured.Unstructured {
```

Comment:
As a side note, we have a dozen or so places in the operator tests where this would be useful and a couple of duplicates of this method. Perhaps we can consolidate in a different PR?